src/gluonts/itertools.py

```diff
@@ -289,6 +289,43 @@ def __iter__(self):
         return filter(self.fn, self.iterable)
 
 
+@dataclass
+class MixIterators:
+    iterators: List[Iterator]
```

Comment:
I think I would make this
```suggestion
    iterators: List[Iterable]
```

Reply:
I talked with Lorenzo specifically on this. If we make it `Iterable`, we will need to create `Iterator`s inside the class, which will be hidden from user. To be explicit, we thought `Iterator` might be better.

Reply:
See my proposal below, I think this should be iterators which we then call `iter` on.

Reply:
> I think this should be iterators which we then call `iter` on

_iterables_